ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -79,178 +86,226 @@ interface State {
+      <div>No runtime exists yet</div>
+      {activeRuntimeOp && <hr/>}
+      {activeRuntimeOp && <div>
+        <ActiveRuntimeOp operation={activeRuntimeOp.operation} workspaceNamespace={workspace.namespace}/>
+      </div>}
+    </React.Fragment>;
+  }
+
+  const isDataproc = (currentRuntime && !!currentRuntime.dataprocConfig);
+  const runtimeChanged = selectedMachine || selectedDiskSize;
+
+  console.log('Status:', currentRuntime, status);
+  return <div data-test-id='runtime-panel'>
+    <h3 style={styles.sectionHeader}>Cloud analysis environment</h3>
+    <div>
+      Your analysis environment consists of an application and compute resources.
+      Your cloud environment is unique to this workspace and not shared with other users.
+    </div>
+    {/* TODO(RW-5419): Cost estimates go here. */}
+    <div style={styles.controlSection}>
+      {/* Recommended runtime: pick from default templates or change the image. */}
+      <PopupTrigger side='bottom'
+                    closeOnClick
+                    content={
+                      <React.Fragment>
+                        <MenuItem style={styles.presetMenuItem}>General purpose analysis</MenuItem>
+                        <MenuItem style={styles.presetMenuItem}>Genomics analysis</MenuItem>
+                      </React.Fragment>
+                    }>
+        <Clickable data-test-id='runtime-presets-menu'
+                   disabled={true}>
+          Recommended environments <ClrIcon shape='caret down'/>
+        </Clickable>
+      </PopupTrigger>
+      <h3 style={styles.sectionHeader}>Application configuration</h3>
+      {/* TODO(RW-5413): Populate the image list with server driven options. */}
+      <Dropdown style={{width: '100%'}}
+                data-test-id='runtime-image-dropdown'
+                disabled={true}
+                options={[currentRuntime.toolDockerImage]}
+                value={currentRuntime.toolDockerImage}/>
+      {/* Runtime customization: change detailed machine configuration options. */}  
+      <h3 style={styles.sectionHeader}>Cloud compute profile</h3>
+      <FlexRow style={{justifyContent: 'space-between'}}>
+        <MachineSelector selectedMachine={selectedMachine} onChange={setselectedMachine} currentRuntime={currentRuntime}/>
+        <DiskSizeSelection selectedDiskSize={selectedDiskSize} onChange={setSelectedDiskSize} currentRuntime={currentRuntime}/>
+      </FlexRow>
+      <FlexColumn style={{marginTop: '1rem'}}>
+        <label htmlFor='runtime-compute'>Compute type</label>
+        <Dropdown id='runtime-compute'
+                  style={{width: '10rem'}}
+                  disabled={true}
+                  options={['Dataproc cluster', 'Standard VM']}
+                  value={isDataproc ? 'Dataproc cluster' : 'Standard VM'}/>
+      </FlexColumn>
+    </div>
+    <FlexRow style={{justifyContent: 'flex-end', marginTop: '.75rem'}}>
+      <Button 
+        disabled={status !== 'Running' || !runtimeChanged}
```

Comment:
I was playing with the Terra app on this - they have an additional "This will make your runtime unavailable" type of confirmation screen in the runtime panel. Once the user confirms they get pushed back to preview mode. The flow seems to make a lot of sense and create a nice UX for the user. Though they do have some bugs wrt runtime state and saving (which I noticed in AoU as well). 

I am thinking of following this UX flow as it answers a lot of questions

Reply:
Note that we have a confirmation page as part of the mocks. I didn't include it explicitly in the scope of this ticket because the update functionality is critical path, and the confirm page is not. However, if you can easily add a simple version of that confirm page here - go for it: https://projects.invisionapp.com/d/main#/console/20179213/431732265/preview

As to what the bugs are, please let me know what these are if you are referencing existing bugs in AoU

Reply:
The bugs were around saving the notebook during, or even after the update cycle. I do not believe AoU has any bug like this in production since there is currently no way to update the runtime.